docs/api/hardhat/hardhat-zksync-upgradable.md

```diff
@@ -0,0 +1,433 @@
+
+
+To use beacon upgradables, start with creating a Deployer for the zkSync network and load the Box artifact:
+
+```typescript
+
+  const testMnemonic = 'stuff slice staff easily soup parent arm payment cotton trade scatter struggle';
+  const zkWallet = Wallet.fromMnemonic(testMnemonic, "m/44'/60'/0'/0/0");
+
+  const deployer = new Deployer(hre, zkWallet);
+
+  const boxContract = await deployer.loadArtifact(contractName);
+
+```
+
+After that, you first need to deploy the beacon contract using `deployBeacon` method from the `zkUpgrades`
+
+```typescript
+  await hre.zkUpgrades.deployBeacon(deployer.zkWallet, boxContract);
+```
+
+In the end, use the `deployBeaconProxy` method which receives the zk wallet, beacon contract and the implementation (Box) contract with its arguments.
+
+```typescript
+  const box = await hre.zkUpgrades.deployBeaconProxy(deployer.zkWallet, beacon, boxContract, [42]);
+```
+After that, your beacon proxy contract is deployed on the network, and you can interact with it.
+
+
+The whole deploy beacon script:
+
+```typescript
+
+import { Deployer } from '@matterlabs/hardhat-zksync-deploy';
+import { Wallet } from 'zksync-web3';
+
+const hre = require('hardhat');
+
+async function main() {
+    const contractName = 'Box';
+    console.log('Deploying ' + contractName + '...');
+
+    const testMnemonic = 'stuff slice staff easily soup parent arm payment cotton trade scatter struggle';
+    const zkWallet = Wallet.fromMnemonic(testMnemonic, "m/44'/60'/0'/0/0");
+
+    const deployer = new Deployer(hre, zkWallet);
+
+    const boxContract = await deployer.loadArtifact(contractName);
+    const beacon = await hre.zkUpgrades.deployBeacon(deployer.zkWallet, boxContract);
+    await beacon.deployed();
+    console.log('Beacon deployed to:', beacon.address);
+
+    const box = await hre.zkUpgrades.deployBeaconProxy(deployer.zkWallet, beacon, boxContract, [42]);
+    await box.deployed();
+    console.log(contractName + ' beacon proxy deployed to: ', box.address);
+
+    box.connect(zkWallet);
+    const value = await box.retrieve();
+    console.log('Box value is: ', value.toNumber());
+}
+
+main();
+
+
+```
+
+
+## Implementation addresses check
+
+Once you deploy the proxy contract, all interactions with your implementation contract go through it. If you invoke the deployProxy function multiple times for a single implementation contract, several proxies will be created, but the implementation contract will remain the same for all of them. This means we can optimize the process to check for the existing implementation addresses before deploying a new proxy, instead of deploying a new implementation contract every time.
```

Comment:
```suggestion
Once you deploy the proxy contract, all interactions with your implementation contract go through it. 

If you invoke the `deployProxy` function multiple times for a single implementation contract, several proxies will be created, but the implementation contract will remain the same for all of them. This means we can optimize the process to check for the existing implementation addresses before deploying a new proxy, instead of deploying a new implementation contract every time.
```